src/Rules/Operators/InvalidComparisonOperationRule.php

```diff
@@ -59,7 +60,9 @@ public function processNode(Node $node, Scope $scope): array
 			return [];
 		}
 
-		if ($this->isNumberType($scope, $node->left) && $this->isNumberType($scope, $node->right)) {
+		$isLeftNumberType = $this->isNumberType($scope, $node->left);
+		$isRightNumberType = $this->isNumberType($scope, $node->right);
+		if (($isLeftNumberType && $isRightNumberType) || (!$isLeftNumberType && !$isRightNumberType)) {
```

Comment:
What about `xor` instead? :D

Reply:
It's fine, thank you for letting me know. I don't think there will be a major issue, PHPStan is continuously tested on several real-world projects (millions of LoC) so I know about the real issues pretty soon.

If something still comes up that a lot of people have problem with, they can wait on a one-before-last version before they upgrade.